mm2src/coins/nft/nft_tests.rs

```diff
@@ -7,9 +7,19 @@ const TEST_WALLET_ADDR_EVM: &str = "0x394d86994f954ed931b86791b62fe64f4c5dac37";
 mod native_tests {
     use crate::nft::nft_structs::{NftTransferHistoryWrapper, NftWrapper, UriMeta};
     use crate::nft::nft_tests::{NFT_HISTORY_URL_TEST, NFT_LIST_URL_TEST, NFT_METADATA_URL_TEST, TEST_WALLET_ADDR_EVM};
-    use crate::nft::send_request_to_uri;
+    use crate::nft::storage::for_db_tests::*;
+    use crate::nft::{check_moralis_ipfs_bafy, send_request_to_uri};
     use common::block_on;
 
+    #[test]
+    fn test_moralis_ipfs_bafy() {
+        let uri =
+            "https://ipfs.moralis.io:2053/ipfs/bafybeifnek24coy5xj5qabdwh24dlp5omq34nzgvazkfyxgnqms4eidsiq/1.json";
+        let res_uri = check_moralis_ipfs_bafy(Some(uri));
+        let expected = "https://ipfs.io/ipfs/bafybeifnek24coy5xj5qabdwh24dlp5omq34nzgvazkfyxgnqms4eidsiq/1.json";
+        assert_eq!(expected, res_uri.unwrap());
+    }
```

Comment:
Please add some malicious testings for `check_moralis_ipfs_bafy` and all other functions that does input validations.

e.g. make the input as `example.com/bafy?1=ipfs.moralis.io&e=https://`

Reply:
added tests: `test_invalid_moralis_ipfs_link`, `test_check_for_spam`

note that function `check_moralis_ipfs_bafy` dont redact url or propagate the error if `token_uri` is invalid.  
This param we get from moralis and if there is unvalid url may be its moralis mistake, or NFT token had this mistake in nft metadata, we should show the uri as it is. We still can try to build `UriMeta` structure from `metadata` field.